### DISCUSSION TITLE
VuePress MarketPlace

### DISCUSSION THREAD

**Discussion by Mister-Hope** [vuepress/core#763] [General]
# 👉 [MarketPlace.VueJS.Press](https://marketplace.vuejs.press)

<img src="https://user-images.githubusercontent.com/33315834/159113267-e4b489ae-4eee-44ec-8103-5fdcd71a3ead.png" align="right" width="350" />

# If you are

- building any plugins and themes for VuePress2
- building any projects or blog with VuePress2
- writing any articles or tutorials about VuePress2

# We welcome you to add your wonderful work to [vuepress-marketplace](https://github.com/vuepress/marketplace)!

## Checklist

We are happy to receive any of them by PRs, but here are some limits.

If you are posting some plugins and themes, you'd better make sure:

- they are not in early W.I.P stage

  To show and tell, you'd better make sure your work is stable enought to use in production, and have a solid API to avoid frequent BREAKING CHANGES in future.

- having basic functionality (plugins) or having basic layouts (themes)

  Your work should be workable and meanful to people reading this list, instead of a piece of test or demo

- having a clear and detailed docs

  People should know how to use them, instead of digging the source code themselves.

**Comment by MicrowaveDev** [comment]
Hello, I'm making IPFS blog engine with VuePress2. It is perfectly fit, as it has a recognizable and simple syntax, can be run from runtime and is well configured, also it is possible to use my own builder.

Here is my [builder](https://github.com/galtproject/vuepress-next/tree/release/packages/%40vuepress/bundler-vite): it's vue bundler-vite fork with additional feature: replacement path of assets with IPFS one.

Motivation of making IPFS blog engine:
Social networks channels bans and important content removals gives me an idea of channel backups with its content and comments. I made auto-backups [module for telegram](https://github.com/galtproject/geesome-node/blob/master/components/socNetClient/telegram/index.ts), and planning to make Instagram and twitter modules also. After channels are backuped to IPFS as data structures - my app needs blog engine module to convert data structures into static site to upload it in IPFS. In general I made a tumblr-like system, where you can create a blog with content management and site representation, but it hosts on IPFS node and can be imported from popular social networks channels.

I made the [MVP of static site generator module](https://github.com/galtproject/geesome-node/tree/master/components/render/static-site-generator) and it works, but need to restart whole app every time after runs VuePress2 module in runtime because of [bug](https://github.com/vuepress/vuepress-next/issues/772). When it will be fixed - I will be glad to share my app in my telegram blog with instructions about how to make full functional site from telegram channel.
